Automattic/jetpack
# Jetpack AI Image: add button entrypoint on the image block

Fixes #38122.

## Proposed changes:

* Add a "Generate with AI" button on the image block, gallery block and slideshow block when the beta flag is enabled (same as the media source menu)
* The button will launch the General Purpose Image Generator and execute all the actions setting the correct placement (a new value, `block-placeholder-button`)
* Make the tool works when the block demands multiple image selection, but keep inserting just one image; this will fix things for the gallery and slideshow blocks

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable beta extensions, go to the block editor and add an image block
* Confirm you can see the "Generate with AI" button:

<img width="500" alt="Screenshot 2024-06-28 at 18 08 24" src="https://github.com/Automattic/jetpack/assets/6760046/04c0a951-4903-4410-a7fb-dd5ab930ae1e">

* Confirm the general purpose image tool will show up when you click the button
* Confirm you can request an image and then use it on the block by clicking "Insert image"
* Do the same for the following blocks:
   * Slideshow
   * Gallery

<img width="500" alt="Screenshot 2024-06-28 at 18 10 37" src="https://github.com/Automattic/jetpack/assets/6760046/719b2f03-8259-48b4-b8cf-d14644090ffe">

* Confirm the following blocks will not show the button:
   * Cover
   * Media & Text
   * Image Compare
   * Story
   * Tiled Gallery
   * these blocks are extended by the same code so they could show the other media sources the code adds, but I opted to limit the usage on them to focus on the core ones
* Confirm the button will not show up when beta extensions are disabled
